ArduinoFrontend/src/app/Libs/CircuitElement.ts

```diff
@@ -410,6 +410,12 @@ export abstract class CircuitElement {
     }
     this.delete();
   }
+  /**
+   * Rotates Component
+   */
+  rotate(): void {
+    this.elements.rotate(90);
```

Comment:
```suggestion
    const bBox = this.elements.getBBox();
    const cx = bBox.x + bBox.width / 2;
    const cy = bBox.y + bBox.height / 2;
    this.elements.rotate(this.rotation, cx, cy);
```

Reply:
all the children elements should be rotated around the center of the parent. else, they will be rotated around their own centre.

Reply:
you can probably write a separate function which returns the centreX and centreY of the element

Reply:
Ok thanks